igor-web/src/main/java/com/netflix/spinnaker/igor/codebuild/AwsCodeBuildAccount.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 Amazon.com, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.spinnaker.igor.codebuild;
+
+import com.amazonaws.auth.AWSStaticCredentialsProvider;
+import com.amazonaws.auth.BasicAWSCredentials;
+import com.amazonaws.services.codebuild.AWSCodeBuildClient;
+import com.amazonaws.services.codebuild.AWSCodeBuildClientBuilder;
+import com.amazonaws.services.codebuild.model.BatchGetBuildsRequest;
+import com.amazonaws.services.codebuild.model.Build;
+import com.amazonaws.services.codebuild.model.StartBuildRequest;
+import lombok.RequiredArgsConstructor;
+
+/** Generates authenticated requests to AWS CodeBuild API for a single configured account */
+@RequiredArgsConstructor
+public class AwsCodeBuildAccount {
+  private final AWSCodeBuildClient client;
+
+  public AwsCodeBuildAccount(String accessKeyId, String secretAccessKey, String region) {
+    BasicAWSCredentials awsCreds = new BasicAWSCredentials(accessKeyId, secretAccessKey);
+    this.client =
+        (AWSCodeBuildClient)
+            AWSCodeBuildClientBuilder.standard()
+                .withCredentials(new AWSStaticCredentialsProvider(awsCreds))
```

Comment:
Ideally you don't use static credentials, you use a role so that you don't have to store static creds in your hal config.  With clouddriver, the 'aws' provider account is configured with a role to assume, and the environment where clouddriver runs is configured with credentials that the SDK default creds provider can pick up to assume that role (like instance creds, ECS creds, env variable creds, etc) using the STSAssumeRoleSessionCredentialsProvider.

So my hal config looks something like this:
```
      accounts:
      - name: my-aws-devel-acct
        requiredGroupMembership: []
        providerVersion: V1
        accountId: '111111111111'
        regions:
        - name: eu-central-1
        assumeRole: role/SpinnakerManaged
```

And then clouddriver just uses the role session provider:
https://github.com/spinnaker/clouddriver/tree/master/clouddriver-aws/src/main/groovy/com/netflix/spinnaker/clouddriver/aws/security

Reply:
The ideal scenario would be specifying aws account name in codebuild account configuration and then all codebuild operations use the credentials from that aws account associated. However, I couldn't find a way to let igor talk to clouddriver to fetch the credentials. Another way is re-implementing the assume role logic in igor. That will introduce duplicate code in two packages. I can do that if that's really necessary.

Reply:
The ideal scenario would be specifying aws account name in codebuild account configuration and then all codebuild operations use the credentials from that aws account associated. However, I couldn't find a way to let igor talk to clouddriver to fetch the credentials. Another way is re-implementing the assume role logic in igor. That will introduce duplicate code in two packages. I can do that if that's really necessary.

Reply:
Yeah agree that would be ideal, but I do think you will need to duplicate the logic unfortunately, though it can probably be simpler in Igor than the way it's implemented in clouddriver